### PR TITLE
[iOS] Stripe Updates - for PaymentIntents & SetupIntents

### PR DESCRIPTION
**Pull request by fbartho**
Work-in-progress fleshing in the iOS bits for the new PaymentIntents & SetupIntents code.

- [x] Prettier-configuration to auto format the JS
- [x] System for Stringy-enums so our keys are a little less vulnerable to copy-paste/spec errors.
- [x] The beginning of some JSDoc comments in the Stripe.js
- [x] Finishing the JSDoc comments
- [x] Initial missing method stubs
- [x] Type-parsers to convert the new parameters into useful Stripe types.
- [x] Implement `confirmPaymentIntent` / `confirmSetupIntent`
- [x] Implement `authenticatePaymentIntent` / `authenticateSetupIntent`
- [x] Test Screens work for iOS too!
- [x] Appium Tests Can Execute

Out-of-scope for this PR -- getting CI working

### CODE REVIEW

File: ios/TPSStripe/TPSStripeManager.m [hunk 1]
```diff
@@ -152,14 +305,252 @@ - (NSDictionary *)constantsToExport
...
+    [api createPaymentMethodWithParams:parsed
+                            completion:^(STPPaymentMethod * __nullable paymentMethod, NSError * __nullable error){
+                                requestIsCompleted = YES;
+
+                                if (error) {
+                                    NSDictionary *jsError = [errorCodes valueForKey:kErrorKeyApi];
+                                    [self rejectPromiseWithCode:jsError[kErrorKeyCode] message:error.localizedDescription];
+                                    return;
+                                }
+                                resolve(paymentMethod);
```
Comment (timrc) [inline ios/TPSStripe/TPSStripeManager.m]: Is this supposed to be converted using `resolve([self convertPaymentMethod:paymentMethod]);` ?

File: ios/TPSStripe/TPSStripeManager.m [hunk 2]
```diff
@@ -152,14 +305,252 @@ - (NSDictionary *)constantsToExport
...
+                                     [self resolvePromise: [self convertConfirmPaymentIntentResult:intent]];
+                                 }
+                             }];
+}
+
+RCT_EXPORT_METHOD(authenticatePayment:(NSDictionary<TPSStripeType(authenticatePayment), id>*)params
+                  resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject) {
+
+    NSString * clientSecret = TPSStripeParam(authenticatePayment, clientSecret);
```
Comment (timrc) [inline ios/TPSStripe/TPSStripeManager.m]: Should this be `NSString * clientSecret = params[TPSStripeParam(confirmPayment, clientSecret)];`

File: ios/TPSStripe/TPSStripeManager.m (hunk 1 above)
Comment (fbartho) [reply ios/TPSStripe/TPSStripeManager.m]: Nice catch! 
```suggestion
                                resolve([self convertPaymentMethod: paymentMethod]);
```

File: ios/TPSStripe/TPSStripeManager.m (hunk 2 above)
Comment (fbartho) [reply ios/TPSStripe/TPSStripeManager.m]: You’re absolutely right, thanks!
```suggestion
    NSString * clientSecret = params[TPSStripeParam(confirmPayment, clientSecret)];
```

File: example/package.json [hunk 3]
```diff
@@ -17,16 +17,22 @@
...
+    "@babel/runtime": "^7.4.5",
     "appium": "1.14.0",
+    "babel-jest": "^24.8.0",
+    "jest": "^24.8.0",
     "metro-react-native-babel-preset": "0.55.0",
+    "react-test-renderer": "16.8.3",
     "tape-async": "2.3.0",
     "tipsi-appium-helper": "3.3.0",
-    "webdriverio": "5.11.7"
+    "webdriverio": "4.14.1"
```
Comment (mindlapse) [inline example/package.json]: No downgrade for webdriver plz :)

**Pull request merged**